driver/bpf/filler_helpers.h

```diff
@@ -826,62 +826,83 @@ static __always_inline int __bpf_val_to_ring(struct filler_data *data,
 		if (!data->curarg_already_on_frame) 
 		{
 			int res;
+			/* Return `res<0` only in case of error. */ 
 			res = bpf_probe_read_str(&data->buf[curoff_bounded], 
 						PPM_MAX_ARG_SIZE,
 						(const void *)val);
-			
-			if (res <= 0)
+			if(res >= 0)
 			{
-				char not_available[] = "<NA>";
-				res = bpf_probe_read_str(&data->buf[curoff_bounded],
-							PPM_MAX_ARG_SIZE,
-							(const void *)not_available);
+				len = res;
+			}
+			else
+			{
+				/* This should be already `0`, but just to be future-proof. */
+				len = 0;
 			}
-			len = res;
 		} 
-		else 
+		else
 		{
 			len = val_len;
 		}
 		break;
 	}
 	case PT_BYTEBUF: {
-		if (data->curarg_already_on_frame || (val && val_len))  
+		if(data->curarg_already_on_frame || (val && val_len))
 		{
 			len = val_len;
 
-			if (enforce_snaplen) {
+			if(enforce_snaplen) 
+			{
 				u32 dpi_lookahead_size = DPI_LOOKAHEAD_SIZE;
 				unsigned int sl;
 
-				if (dpi_lookahead_size > len)
+				if(dpi_lookahead_size > len)
+				{
 					dpi_lookahead_size = len;
+				}
 
-				if (!data->curarg_already_on_frame) {
+				if(!data->curarg_already_on_frame) 
+				{
+					/* We need to read the first `dpi_lookahead_size` bytes. 
+					 * If we are not able to read at least `dpi_lookahead_size` 
+					 * we send an empty param `len=0`.
+					 */
 					volatile u16 read_size = dpi_lookahead_size;
 
 #ifdef BPF_FORBIDS_ZERO_ACCESS
-					if (read_size)
-						if (bpf_probe_read(&data->buf[curoff_bounded],
-								   ((read_size - 1) & SCRATCH_SIZE_HALF) + 1,
-								   (void *)val))
+					if(!read_size || bpf_probe_read(&data->buf[curoff_bounded],
```

Comment:
I see `read_size` is now negated and the logic seems sound. Was this broken before?

Reply:
that's a good question, no the logic was not wrong it's just a refactor we can say, check `read-size` is just a trick to please the verifier, as you can notice it will be never `0`, but it will be always `DPI_LOOKAHEAD_SIZE`

Reply:
the previous logic was: (2 steps)
* check that there is something to read  `if(read_size)`
* if yes, read it and in case of failure return `PPM_FAILURE_INVALID_USER_MEMORY`

Now it is: (just 1 step)
* if there is nothing to read (!read_size) or the actual read operation fails, return an empty param.

the negation is the result of just merging the 2 steps, i don't know if it is clear :thi

Reply:
logically it is not the same thing but considering that `read_size` is always `0` it is the same stuff